### PR TITLE
Consensus dependent antiflood parameters

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added flood preventer adapting values capability against consensus size

### CODE REVIEW

File: process/throttle/antiflood/factory/p2pOutputAntiflood.go [hunk 1]
```diff
@@ -28,15 +28,20 @@ func initP2POutputAntiFlood(mainConfig config.Config) (process.P2PAntifloodHandl
...
-		antifloodCache,
-		make([]floodPreventers.QuotaStatusHandler, 0),
-		peerMaxMessagesPerSecond,
-		peerMaxTotalSizePerSecond,
-		outputReservedPercent,
-	)
+	basePeerMaxMessagesPerInterval := mainConfig.Antiflood.PeerMaxOutput.BaseMessagesPerInterval
+	peerMaxTotalSizePerInterval := mainConfig.Antiflood.PeerMaxOutput.TotalSizePerInterval
+	arg := floodPreventers.ArgQuotaFloodPreventer{
+		Name:                      "output",
```
Comment (sasurobert) [inline process/throttle/antiflood/factory/p2pOutputAntiflood.go:34]: magic string

File: process/smartContract/hooks/blockChainHook.go [hunk 2]
```diff
@@ -179,7 +179,17 @@ func (bh *BlockChainHookImpl) GetStorageData(accountAddress []byte, index []byte
 	}
 
 	value, err := userAcc.DataTrieTracker().RetrieveValue(index)
-	log.Trace("GetStorageData ", "address", accountAddress, "rootHash", userAcc.GetRootHash(), "key", index, "value", value, "error", err)
+	messages := []interface{}{
```
Comment (raduchis) [inline process/smartContract/hooks/blockChainHook.go:182]: interesting resolve of the if +1

File: process/throttle/antiflood/floodPreventers/quotaFloodPreventer.go [hunk 3]
```diff
@@ -25,60 +38,75 @@ type quota struct {
...
 // quotaFloodPreventer represents a cache of quotas per peer used in antiflooding mechanism
 type quotaFloodPreventer struct {
-	maxMessagesPerPeer uint32
-	mutOperation       *sync.RWMutex
-	cacher             storage.Cacher
-	statusHandlers     []QuotaStatusHandler
-	maxSizePerPeer     uint64
-	percentReserved    uint32
+	name                          string
+	mutOperation                  *sync.RWMutex
```
Comment (raduchis) [inline process/throttle/antiflood/floodPreventers/quotaFloodPreventer.go:42]: is the pointer to the RWMutex necessary?

Comment (iulianpascalau) [reply process/throttle/antiflood/floodPreventers/quotaFloodPreventer.go:42]: no, it was necessary in the old implementation in which the p2p antiflood struct was constructed above quotaFloodPreventer. Will remove the pointer ASAP.

File: process/throttle/antiflood/factory/p2pOutputAntiflood.go (hunk 1 above)
Comment (iulianpascalau) [reply process/throttle/antiflood/factory/p2pOutputAntiflood.go:34]: will change

Comment (iulianpascalau) [reply process/throttle/antiflood/factory/p2pOutputAntiflood.go:34]: done

File: process/throttle/antiflood/floodPreventers/quotaFloodPreventer.go (hunk 3 above)
Comment (iulianpascalau) [reply process/throttle/antiflood/floodPreventers/quotaFloodPreventer.go:42]: done

**Pull request merged**